### PR TITLE
Feat: Add reviews and ratings features

### PR DESCRIPTION
**Pull request by Vansh353**

### CODE REVIEW

File: dtos/product_models.py [hunk 1]
```diff
@@ -22,4 +19,33 @@ class ProductSearch(BaseModel):
...
     discount: Optional[float] = None
     rating: Optional[float] = None
-    sort_by: Optional[str] = None
\ No newline at end of file
+    sort_by: Optional[str] = None
+    
+class ProductReview(BaseModel):
+    
+    rating: Optional[float] 
+    review: Optional[str] = None
```
Comment (niyati-doshi20) [inline dtos/product_models.py]: Make both field compulsory

Comment (Vansh353) [reply dtos/product_models.py]: okay done

Comment (Vansh353) [reply dtos/product_models.py]: any other changes?

File: controllers/product_controller.py [hunk 2]
```diff
@@ -72,14 +84,98 @@ def get_product_categories():
...
+            product_table.c.category,
+            product_table.c.price,
+            product_table.c.description,
+            product_table.c.discount,
+            review_table.c.user_id,
+            users_table.c.name,  # Group by the user name
+            review_table.c.review,
+            review_table.c.rating
+        )
+        query = query.where(product_table.c.id == product_id)
```
Comment (niyati-doshi20) [inline controllers/product_controller.py]: Do not give same variable name

Comment (Vansh353) [reply controllers/product_controller.py]: are u talking about the query i used at the start and with the where?

Comment (niyati-doshi20) [reply controllers/product_controller.py]: Yes

File: controllers/product_controller.py [hunk 3]
```diff
@@ -72,14 +84,98 @@ def get_product_categories():
...
+            id=product[0],
+            name=product[1],
+            category=product[2],
+            price=product[3],
+            description=product[4],
+            rating=average_rating,  # Use the calculated average rating,
+            discount=product[5],
+            reviews=reviews,
+        )
+
```
Comment (niyati-doshi20) [inline controllers/product_controller.py]: Do it without indexing.

**Pull request merged**